infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -97,25 +100,81 @@ public LogReplicationMetadataManager(CorfuRuntime runtime, LogReplicationContext
-
-            this.replicationEventTable = this.corfuStore.openTable(NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
-                    ReplicationEventInfoKey.class,
-                    ReplicationEvent.class,
-                    null,
-                    TableOptions.fromProtoSchema(ReplicationEvent.class));
-        } catch (Exception e) {
-            log.error("Caught an exception while opening metadata tables", e);
-            throw new ReplicationWriterException(e);
+            table = corfuStore.openTable(namespace, tableName,
+                    keyClass, valueClass, metadataClass,
+                    TableOptions.fromProtoSchema(valueClass));
+        } catch (Exception ee) {
+            log.error("Caught an exception while opening "+namespace+"$"+tableName, ee);
+            throw new ReplicationWriterException(ee);
         }
+        return table;
+    }
+
+    /**
+     * The following tables need to move from old type to new type
+     * 1. CORFU-REPLICATION-WRITER-UUID-OF-CLUSTER -> LogReplicationMetadataTable
+     * Type: Schema change AND name change
+     *    (LogReplicationMetadataKey, LogReplicationMetadataVal) ->
+     *     (LogReplicationSession, ReplicationMetadata)
+     * Action: Open the old table with old types and Clear the old table
+     *         Not used in new code. Optional as failure to clear won't cause issues.
+     *         TODO: consider scanning registry for signature and wipe out contents.
+     *
+     * 2. LogReplicationStatus
+     * Type: Schema change for key AND value types with NO table name change
+     * (ReplicationStatusKey, ReplicationStatusVal) ->
+     * (LogReplicationSession.class, ReplicationMetadata.class)
+     * Action: Make serializer aware of old types, open table with new type, clear it
+     *
+     * 3. LogReplicationEventTable
+     * Type: Schema change for Key type only
+     * (ReplicationEventKey, ReplicationEvent) ->
+     * (ReplicationEventInfoKey, ReplicationEvent)
+     * Action: Make serializer aware of old key type, open table with new type, clear it
+     * @param txnContext - the parent transaction where check for rolling upgrade completion occurs
+     */
+    public static void migrateData(CorfuStore corfuStore, TxnContext txnContext) {
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationStatusKey.getDefaultInstance());
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationStatusVal.getDefaultInstance());
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationEventKey.getDefaultInstance());
+        executorService.submit(() -> {
+            tryOpenTable(corfuStore, NAMESPACE, LogReplicationUtils.REPLICATION_STATUS_TABLE_NAME,
```

Comment:
unable to follow why we open the tables in the new schema here?

Reply:
we need to open the table to clear it - so we have to use some schema right? The choice here is to NOT open it in old schema to avoid polluting the registry table with stale values.